<a href="https://colab.research.google.com/github/jampsss/Probabilidad-yEstadistica/blob/main/Unidad2/Probabilidad_U2_Programa3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Probabilidad y Estadistica
# **Unidad 2**
Docente: *Dr Jose Gabriel Rodriguez Rivas*
Alumno: *Luis Enrique Lerma Romero*




In [1]:
import math

# Definir los parámetros del problema
n = 10  # Total de asistentes disponibles
r = 4   # Preguntas a calificar (asistentes a seleccionar)

# Método 1: Cálculo paso a paso usando factoriales
factorial_n = math.factorial(n)
factorial_resta = math.factorial(n - r)
permutaciones_paso_a_paso = factorial_n // factorial_resta

# Método 2: Función directa de permutaciones de math
permutaciones_directa = math.perm(n, r)

# Mostrar resultados
print(f"Total de asistentes (n): {n}")
print(f"Preguntas a calificar (r): {r}")
print("---")
print(f"10! = {factorial_n}")
print(f"(10 - 4)! = 6! = {factorial_resta}")
print(f"Resultado paso a paso ({factorial_n} / {factorial_resta}): {permutaciones_paso_a_paso}")
print(f"Resultado directo con math.perm(10, 4): {permutaciones_directa}")

Total de asistentes (n): 10
Preguntas a calificar (r): 4
---
10! = 3628800
(10 - 4)! = 6! = 720
Resultado paso a paso (3628800 / 720): 5040
Resultado directo con math.perm(10, 4): 5040


### Ejercicio: Permutaciones de asistentes
Se tienen $n = 10$ asistentes para calificar $r = 4$ preguntas distintas (un asistente por pregunta).

$$P(n, r) = \frac{n!}{(n - r)!} = \frac{10!}{(10 - 4)!} = \frac{10!}{6!} = \frac{3\,628\,800}{720} = 5040$$